### DISCUSSION TITLE
Regular Regression vs. Ordinal Regression Loss

### DISCUSSION THREAD

**Discussion by abarbosa94** [Raschka-research-group/coral-pytorch#40] [Q&A] [answered]
I just watched this [presentation](https://www.youtube.com/watch?v=E--9kKauJkM) and really liked it; kudos!

This might be a dummy question, but I asked myself, if you're using MAE metrics to compute model performance, why not train the whole model using MSE Loss or MAE Loss and then rounding to the nearest integer at inference time? Is it to preserve the probabilities when making predictions?

Is there any theoretical foundation that justifies that? Please let me know if this is covered in any of the CORN or CORAL papers

Thanks!

**Comment by rasbt** [comment, marked as answer]
Good point, and you can absolutely do that @abarbosa94 . But when you train a regression model minimize MSE during training you would assume that all the distances are equal. Like the difference between labels 3 and 5 is the same as the difference between labels 5 and 7. That might not be the case in ordinal regression though.

**Reply by abarbosa94** [reply]
Thank you so much!
